pkg/controller/scyllacluster/resource.go

```diff
@@ -1125,3 +1190,270 @@ func MakeJobs(sc *scyllav1.ScyllaCluster, services map[string]*corev1.Service, i
+	var endpointSlices []*discoveryv1.EndpointSlice
+	var endpoints []*corev1.Endpoints
+
+	endpointPortsHash := func(name string, ports []discoveryv1.EndpointPort) (string, error) {
+		h, err := hash.HashObjects(name, ports)
+		if err != nil {
+			return h, err
+		}
+
+		return hex.EncodeToString([]byte(h)[0:])[:8], nil
+	}
+
+	for _, svc := range services {
+		if svc.Labels[naming.ScyllaServiceTypeLabel] != string(naming.ScyllaServiceTypeMember) {
+			continue
+		}
+
+		endpointSliceLabels := maps.Clone(svc.Labels)
+		endpointSliceLabels[discoveryv1.LabelManagedBy] = naming.OperatorAppName
+		endpointSliceLabels[discoveryv1.LabelServiceName] = svc.Name
+
+		endpointsLabels := maps.Clone(svc.Labels)
+		endpointsLabels[discoveryv1.LabelSkipMirror] = "true"
+
+		pod, err := podLister.Pods(sc.Namespace).Get(svc.Name)
+		if err != nil {
+			if apierrors.IsNotFound(err) {
+				continue
+			}
+			return endpointSlices, endpoints, fmt.Errorf("can't get Pod %q: %w", naming.ManualRef(sc.Namespace, svc.Name), err)
+		}
+
+		// Don't publish endpoints for deleted Pod (#1077)
+		if pod.DeletionTimestamp != nil {
+			continue
+		}
+
+		podReference := &corev1.ObjectReference{
+			Kind:      "Pod",
+			Namespace: pod.Namespace,
+			Name:      pod.Name,
+			UID:       pod.UID,
+		}
+
+		var podPorts []discoveryv1.EndpointPort
+		containerPortMap := map[string][]discoveryv1.EndpointPort{}
+
+		for _, servicePort := range svc.Spec.Ports {
+			ep := discoveryv1.EndpointPort{
+				Name:        pointer.Ptr(servicePort.Name),
+				Port:        pointer.Ptr(servicePort.Port),
+				Protocol:    pointer.Ptr(servicePort.Protocol),
+				AppProtocol: servicePort.AppProtocol,
+			}
+
+			var err error
+
+			containerServingPort, _, ok := slices.Find(pod.Spec.Containers, func(c corev1.Container) bool {
+				_, _, ok := slices.Find(c.Ports, func(cp corev1.ContainerPort) bool {
+					if servicePort.Name == cp.Name {
+						return true
+					}
+
+					if servicePort.Port == cp.ContainerPort {
+						return true
+					}
+
+					if servicePort.TargetPort.String() != "" {
+						switch servicePort.TargetPort.Type {
+						case intstr.String:
+							return cp.Name == servicePort.TargetPort.StrVal
+						case intstr.Int:
+							return cp.ContainerPort == servicePort.TargetPort.IntVal
+						default:
+							err = fmt.Errorf("unsupported type of intstr.IntOrString %d in %q Service", servicePort.TargetPort.Type, naming.ObjRef(svc))
+						}
+					}
+
+					return false
+				})
+				return ok
+			})
+			if err != nil {
+				return endpointSlices, endpoints, fmt.Errorf("can't find container serving port: %w", err)
+			}
+
+			// Container serves given Service port and decides about readiness
+			if ok && containerServingPort.ReadinessProbe != nil {
+				containerPortMap[containerServingPort.Name] = append(containerPortMap[containerServingPort.Name], ep)
```

Comment:
shouldn't you check for collisions and at least panic then? Assuming the output hash of that entire composition fulfills bit independence criterion, you have 2^32 chance of hitting it taking the first 8 chars. Not that high, but...

Reply:
I aligned code to lower chance of collision and added unit test verifying that spec of service and statefulset doesn't generate any collisions.

Reply:
multiple ports can be handled by single container, so it's expected there's a collision here.